### PR TITLE
Fix #135: better GPS point validator detection of invalid coordinate numeric conversions 

### LINKED ISSUE #135: GpsPoint validator needs a rewrite because it returns true for the input "foo, bar"
> I believe this validator should either be rewritten or removed entirely as it will return true for the input "foo, bar"
> 
> It expects a string as input in the form "latitude, longitude" :
> 
> ```
>     public function isValid($value)
>     {
>         if (strpos($value, ',') === false) {
>             $this->error(self::INCOMPLETE_COORDINATE, $value);
>             return false;
>         }
> 
>         [$lat, $long] = explode(',', $value);
> ```
> 
>  but the `isValidCoordinate()` method blindly casts the supposed lat or long to a double:
> 
> ```
>         $doubleLatitude = (double) $value;
> 
>         if ($doubleLatitude <= $maxBoundary && $doubleLatitude >= $maxBoundary * -1) {
>             return true;
>         }
> ```
> 
> which results in a 0 value in the case of non-numerical strings, which then passes validation as a valid lat/long.
> 
> At the very least it should have explicit comments stating that the expected input is in a very particular format already.

_Originally posted by @ajgale in https://github.com/laminas/laminas-validator/issues/69#issuecomment-1162331488_

### PR DESCRIPTION
**Pull request by codisart**
|    Q          |   A
|-------------- | ------
| Bugfix        | yes
| BC Break      | yes

### Description

This PR is trying to fix the issue #135. 
The fix is to have the string "foo" not converted to 0.0 and validated as a correct coordinate.

### CODE REVIEW

File: src/GpsPoint.php [hunk 1]
```diff
@@ -63,19 +53,17 @@ private function isValidCoordinate($value, float $maxBoundary): bool
             $value = $this->removeDegreeSign($value);
         }
 
-        if ($value === false || $value === null) {
+        if ($value === false || $value === null || !is_numeric($value)) {
```
Comment (Ocramius) [inline src/GpsPoint.php]: This ternary can be simplified to just the `is_numeric()` check

File: test/GPSPointTest.php [hunk 2]
```diff
@@ -66,7 +67,7 @@ public function testErrorsSetOnOccur(string $value, string $messageKey, string $
     }
 
     /**
-     * @psalm-return array<array-key, array{0: string}>
+     * @psalm-return array<array-key, arrayG{0: string}>
```
Comment (Ocramius) [inline test/GPSPointTest.php]: A `G` slipped in

**Review by Ocramius — CHANGES_REQUESTED** [review]
Please check CI failures too

**Comment by Ocramius** [comment]
As this is a bugfix, it should target `2.29.x`

File: test/GPSPointTest.php [hunk 3]
```diff
@@ -85,9 +86,11 @@ public function basicDataProvider(): array
     public function errorMessageTestValues(): array
     {
         return [
-            ['63 47 24.691 N, 18 2 54.363 W', GpsPoint::OUT_OF_BOUNDS, '63 47 24.691 N'],
+            ['63 47 24.691 N, 18 2 54.363 W', GpsPoint::CONVERT_ERROR, '63 47 24.691 N'],
+            ['65° 4\' N,-22.728867530822754', GpsPoint::CONVERT_ERROR, '65° 4\' N'],
```
Comment (froschdesign) [inline test/GPSPointTest.php]: The previous test was fine, so out of bounds cannot be a conversion error now!

Comment (froschdesign) [reply test/GPSPointTest.php]: See my comment on the related issue report: https://github.com/laminas/laminas-validator/issues/135#issuecomment-1165290208

File: test/GPSPointTest.php (hunk 2 above)
Comment (codisart) [reply test/GPSPointTest.php]: Thank you, I removed it

File: test/GPSPointTest.php (hunk 3 above)
Comment (codisart) [reply test/GPSPointTest.php:89]: Hello @froschdesign, this change in the provider is a mistake for this PR.
I have a new fix where I use the casting to float to determine is the previous string was a partial number.

Comment (codisart) [reply test/GPSPointTest.php]: Hello @froschdesign, this change in the provider is a mistake for this PR.
I have a new fix where I use the casting to float to determine is the previous string was a partial number before the casting.

File: src/GpsPoint.php (hunk 1 above)
Comment (codisart) [reply src/GpsPoint.php]: After the comment of froschdesign, this check has became obsolete.

**Review by Ocramius — APPROVED** [review]
Thanks @codisart!

**Pull request merged**